### DISCUSSION TITLE
How to restrict execution of GitHub Actions workflow on tags creation only

### DISCUSSION THREAD

**Discussion by glambert** [community/community#24925] [Actions]
<p>Hey guys!</p>
<p>Been using GitHub Actions v2 (CI/CD) to implement publishing of npm packages when a new tag is pushed to our GitHub repo. I have looked into <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-restricting-the-workflow-run-to-specific-refs-and-paths" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-restricting-the-workflow-run-to-specific-refs-and-paths</a> but can’t get it to work. Here’s a snippet of my workflow file:</p>
<pre><code>on:
  push:
    tags:
    - my-tag
...
</code></pre>
<p>Problem is when I push a new commit even with no tags it still gets triggered. I also tried using the <code>create</code> event but still no dice, when a new branch is created it still triggers the workflow.</p>
<p>Am I doing something wrong with filters? Would you have an alternative perhaps on how to trigger a workflow only when a new tag is created?</p>
<p>Also, if I want to filter tags that have a @ sign, is that the way to do it?</p>
<pre><code>on:
  push:
    tags:
    - \@orgname/*
</code></pre>
<p>Which should mean "Only run on tags that start with <code>@orgname/</code>".</p>
<p>Thanks!</p>

**Comment by PilotConway** [comment, marked as answer]
<p>I was just running into this. I solved just using a full ref path: </p>
<pre><code>on:
  push:
    branches:
      - refs/tags/*
</code></pre>
<p>So I’m pretty sure you can then restrict to scopes like this: </p>
<pre><code>on:
  push:
    branches:
      - refs/tags/@orgname/*
</code></pre>
<p>I basically wanted to wildcard any tag push only, but this is the only way I could get it to work right.</p>

**Comment by glambert** [comment]
<p>Should it be with <code>tags</code> instead of <code>branches</code>?</p>
<pre><code>on:
  push:
    tags:
      - refs/tags/*
</code></pre>
<p>Will give this a try, thanks!</p>

**Comment by PilotConway** [comment]
<p>Tags is just shorthand for <code>refs/tags</code> so branches: <code>refs/tags/*</code> is the equivalent and will do exactly what you wanted. I’ve been testing it for a bit this afternoon and it’s working.</p>
<p>My guess is <code>tags: refs/tags/*</code> might try to do something like <code>refs/tags/refs/tags</code> but not entirely sure. It might work. :shrug:</p>

**Comment by glambert** [comment]
<pre><code>on:
  push:
    branches:
      - refs/tags/@orgname/*
</code></pre>
<p>Worked like a charm, thanks a lot  <a class="mention" href="/pilotconway">@pilotconway</a> !</p>

**Comment by helen** [comment]
<p>I just ran into this myself - seems like a bug in Actions itself? The branches workaround seems fine for now but not properly descriptive IMO.</p>

**Comment by PilotConway** [comment]
<p>I agree, I have a couple work flows I really want to do something like, on _any_ tag push or these specific branches, but the docs don’t have a good way to mix and match like that either. I’m not sure if its not enough documentation or a lack of understanding on yaml on my part. </p>

**Comment by glambert** [comment]
<p>In fact it seems it works half and half, as in:</p>
<pre><code>on:
  push:
    branches:
      - refs/tags/@orgname/*
</code></pre>
<p>will only trigger when new tag is pushed,  <strong>but</strong> , if I push a tag named _test-tag _for example it will still trigger the workflow. What I’m wondering is if this “filter” looks at all  <strong>existing </strong> tags in refs/ and thus will match if you already have any “<span class="mention">@orgname</span>/*” tag present.</p>
<p>Still confused by this whole workflow execution restriction thing, but hey its still beta :wink:</p>

**Comment by jeremyepling** [comment]
<p>This is a bug that should be fixed this week. It should only trigger when a tag is pushed.</p>

**Comment by stefanluptak** [comment]
<p>That are wonderful news. Thanks a lot!</p>

**Comment by SergeyLadutko** [comment]
<p>did you manage to fix this error ?</p>

**Comment by leachim** [comment]
<p>This is still an issue for me, using the following syntax:</p>
<pre><code>on:
  push:
    branches:
      - master
      - refs/tags/*
jobs:
  build:
</code></pre>
<p>and I also tried the other suggestions. Doesn’t seem to work. I would like to build only for commits that are tagged. </p>

**Comment by rabc** [comment]
<p>Around here I can see the correct behavior (only execute only when the tag is pushed) when I fork the repo to my personal account, but it is triggered for every push on the organization’s repo.</p>

**Comment by rabidkitten** [comment]
<p>This doesn’t work. It runs on every single commit regardless of whether the commit was tagged. Double-quotes, single-quotes - doesn’t matter.</p>
<p>on:<br>
push:<br>
branches:<br>
- main<br>
tags:<br>
- ‘v*.<em>.</em>’</p>

**Reply by jeacott1** [reply]
I suspect whats happening here is that its triggering everything on branches: main OR tags: 'v*' which is not the same as other things - which is awful. a workaround is to use something like if: github.ref == 'refs/heads/whatever'
